MPM/mpm.php

```diff
@@ -96,8 +96,10 @@ function mpm_webhook()
 		$order_id	= $payment->metadata->order_id;
 		$order		= new WC_Order( $order_id );
 
-		// Don't update an already paid order
-		if (!$order->needs_payment())
+		$onHold = $order->get_status() == 'on-hold';
+
+		// Don't update an already paid order, needs_payment() searches by valid_statusses, but does not include on-hold status, so we add it here.
+		if (!$order->needs_payment() && !$onHold)
```

Comment:
Ik ben het met je eens. Alsof een on-hold status geen betaling nodig heeft. Ik denk dat het ook in de business logica bedoeld is voor een handmige status wijziging bij on-hold orders, want een bankoverschrijving wordt ook met de hand gewijzigd naar betaald, maar dat verzin ik zo uit de lucht ;) 

 

De get_status() ga ik voor de versies goed fixen. 

 

From: Lennard van Gunst [mailto:notifications@github.com] 
Sent: woensdag 15 april 2015 17:14
To: mollie/WooCommerce
Cc: George van Engers
Subject: Re: [WooCommerce] Banktransfer orders are placed on-hold. When payment expires, or fails, ... (#27)

 

In MPM/mpm.php <https://github.com/mollie/WooCommerce/pull/27#discussion_r28428818> :

> @@ -96,8 +96,10 @@ function mpm_webhook()
>              $order_id      = $payment->metadata->order_id;
>              $order         = new WC_Order( $order_id );
>  
> -            // Don't update an already paid order
> -            if (!$order->needs_payment())
> +            $onHold = $order->get_status() == 'on-hold';
> +
> +            // Don't update an already paid order, needs_payment() searches by valid_statusses, but does not include on-hold status, so we add it here.
> +            if (!$order->needs_payment() && !$onHold)

Raar dat needs_payment() alleen pending of failed checked. Maar in payment_complete() kan je wel de betaling afronden als status cancelled of on-hold is https://github.com/woothemes/woocommerce/blob/master/includes/abstracts/abstract-wc-order.php#L2208.

—
Reply to this email directly or view it on GitHub <https://github.com/mollie/WooCommerce/pull/27/files#r28428818> .  <https://github.com/notifications/beacon/ADrkExYN3dSs2T_M0pZuaKh5MAo9aLRNks5n_ne0gaJpZM4EA-tn.gif>